src/mbgl/style/expression/value.cpp

```diff
@@ -136,6 +143,31 @@ mbgl::Value ValueConverter<mbgl::Value>::fromExpressionValue(const Value& value)
             assert(false);
             return mbgl::Value();
         },
+        [&](const Formatted& formatted)->mbgl::Value {
+            // TODO format: figure out right way to serialize this?
```

Comment:
😳I missed that I had left these in! There are actually three linked TODOs and I could use some help thinking them through.

- `stringify(Writer& writer, const expression::Formatted& v)` writes a `Formatted` literal as the expression that would generate that literal, since there's no other way to represent a `Formatted` object.
- `value.cpp`'s `writeJSON` just calls through to the implementation in `stringify.hpp`, which I think makes sense, but I'm not sure if there's some context I'm missing about how that's supposed to work, because it seems like most of `writeJSON` is duplicating code in `stringify.hpp`
- The `(const Formatted& formatted)->mbgl::Value` converter does the same thing as the `stringify` implementation, except it's in the `mbgl::Value` format instead of writing out to a JSON writer. This seems like bad duplication. I'm somewhat following the pattern of `Color` (which has an `mbgl::Value` serialization and also a parallel `stringify`), but the duplication seems worse since the logic for `Formatted` is more complicated.

Maybe the right way to do this is to give `Formatted` a method that serializes it to an `mbgl::Value`, and then use that for all three situations (since there's already an `mbgl::Value` -> JSON serializer)?